cpp/src/arrow/python/deserialize.cc

```diff
@@ -442,9 +451,15 @@ Status GetSerializedFromComponents(int num_tensors,
 
     std::shared_ptr<SparseTensor> sparse_tensor;
     ARROW_ASSIGN_OR_RAISE(sparse_tensor, ipc::internal::ReadSparseTensorPayload(payload));
+    csf_index_buffers += 2 * sparse_tensor->ndim() - 1;
```

Comment:
I guess this should be calculated only for CSF tensors.

Reply:
Fixed.